.eslintrc.js

```diff
@@ -13,6 +13,11 @@ module.exports = {
     'jsx-a11y/click-events-have-key-events': 'off',
     'jsx-a11y/no-static-element-interactions': 'off',
     'jsx-a11y/anchor-is-valid': 'off',
+    '@typescript-eslint/no-use-before-define': 'off',
```

Comment:
Dla mnie wyłączenie tej reguły dla całego projektu nie jest dobrym pomysłem. Porównałbym to do stosowania `any` gdzie potem wartość TS spada z każdym kolejnym `any`. Nie bez powodu deklaracja powinna być przed wykorzystaniem zmiennej, a jeżeli juz jest przypadek gdzie chcesz świadomie złamać tą regułę to można zrobić to jednorazowo w wybranym miejscu, zamiast wyłączać to dla całego projektu.